compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2395,7 +2395,12 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
 
     object Implicits extends ImplicitsModule:
       def search(tpe: TypeRepr): ImplicitSearchResult =
-        ctx.typer.inferImplicitArg(tpe, Position.ofMacroExpansion.span)
+        import tpd.TreeOps
+        val implicitTree = ctx.typer.inferImplicitArg(tpe, Position.ofMacroExpansion.span)
+        // Make sure that we do not have any uninstantiated type variables.
+        // See tests/pos-macros/exprSummonWithTypeVar with -Xcheck-macros.
+        implicitTree.foreachSubTree(tree => dotc.typer.Inferencing.fullyDefinedType(tree.tpe, "", tree))
```

Comment:
foreachSubTree is expensive.  You could use typer.interpolateTypeVars instead (probably with `pt = WildcardType` and `locked = ctx.typerState.ownedVars`).

Reply:
`interpolateTypeVars` did not work.

Reply:
I there a way to make `inferImplicitArg` resolve all type variables that it inserts?

Reply:
Try defining val locked = ctx.typerState.ownedVars before the call to inferImplicitArg, so that variables created by it can be instantiated when you pass locked to interpolateTypeVars

Reply:
It did not work either. Note that the `ownedVars` of this context are empty before and after the call to `inferImplicitArg`. The type variable in introduced within `inferImplicitArg`.